dataset.go

```diff
@@ -23,6 +23,8 @@ var ErrorElementNotFound = errors.New("element not found")
 // within this Dataset (including Elements nested within Sequences).
 type Dataset struct {
 	Elements []*Element `json:"elements"`
+
+	opts parseOptSet
```

Comment:
I can take care of this in another PR, but just putting a note for myself: I think if we're going to add this set of options to Dataset, it makes sense that any person who holds a Dataset should be able to inspect the `opts ...ParseOption` the Dataset was parsed with. Typically I don't like just tacking on options to another struct to "pass it through" but in this scenario I think this makes sense because it seems useful for anyone who has a Dataset to see the options it was parsed with.